registrations/views.py

```diff
@@ -137,3 +156,109 @@ def get(self, request, *args, **kwargs):
+                            'addresses': {
+                                'msisdn': {
+                                    hcw_msisdn: {'default': True}
+                                }
+                            }
+                        }
+                    }
+                    hcw_identity = is_client.create_identity(identity)
+                else:
+                    hcw_identity = result['results'][0]
+            else:
+                hcw_identity = None
+
+            id_type = serializer.validated_data['mom_id_type']
+            if hcw_identity is not None:
+                operator = hcw_identity['id']
+                device = hcw_msisdn
+            else:
+                operator = None
+                device = mom_msisdn
+            # Get or create Mom Identity
+            result = is_client.get_identity_by_address(
+                'msisdn', mom_msisdn)
+            if 'results' in result and not result['results']:
+                identity = {
+                    'details': {
+                        'default_addr_type': 'msisdn',
+                        'addresses': {
+                            'msisdn': {
+                                mom_msisdn: {'default': True}
+                            }
+                        }
+                    },
+                    'operator_id': operator,
+                    'lang_code': lang_code,
+                    'id_type': id_type,
+                    'mom_dob': serializer.validated_data['mom_dob'],
+                    'edd': serializer.validated_data['mom_edd'],
+                    'faccode': serializer.validated_data['clinic_code'],
+                    'consent': serializer.validated_data['consent'],
+                    'mha': serializer.validated_data['mha'],
+                    'swt': serializer.validated_data['swt'],
+                }
+                if id_type == 'sa_id':
+                    identity['details']['sa_id_no'] = (
+                        serializer.validated_data['mom_id_no'])
+                elif id_type == 'passport':
+                    identity['details']['passport_origin'] = (
+                        serializer.validated_data['mom_passport_origin'])
+                    identity['details']['passport_no'] = (
+                        serializer.validated_data['mom_id_no'])
+                mom_identity = is_client.create_identity(identity)
+            else:
+                mom_identity = result['results'][0]
+
+            # Create registration
+            reg_data = {
+                'operator_id': operator,
+                'msisdn_registrant': mom_msisdn,
+                'msisdn_device': device,
+                'id_type': id_type,
+                'language': lang_code,
+                'edd': serializer.validated_data['mom_edd'],
+                'faccode': serializer.validated_data['clinic_code'],
+                'consent': serializer.validated_data['consent'],
+
```

Comment:
`reg_data` will also depend on the `authority` in `serializer.validated_data`.  To map to the new system registrations:
clinic = hw_full
chw = hw_partial
personal/public = patient